xbmc/ContextMenuItem.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ *      Copyright (C) 2015 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ContextMenuItem.h"
+#include "addons/Addon.h"
+#include "addons/AddonManager.h"
+#include "addons/ContextMenuAddon.h"
+#include "addons/IAddon.h"
+#include "interfaces/generic/ScriptInvocationManager.h"
+#include "interfaces/python/ContextItemAddonInvoker.h"
+#include "interfaces/python/XBPython.h"
+#include "utils/StringUtils.h"
+#include <boost/lexical_cast.hpp>
+
+
+std::string CContextMenuItem::GetLabel() const
+{
+  if (!m_addon)
+    return "";
+
+  if (StringUtils::IsNaturalNumber(m_label))
+  {
+    return m_addon->GetString(boost::lexical_cast<int>(m_label.c_str()));
```

Comment:
@Paxxi and others will yell at you for sure because of the `boost::lexical_cast` ;) I can recall there was some talk (or even a proposal) for a c++11 alternative. Not sure if it ever happened though.